scripts/process/process.classified_precip.R

```diff
@@ -0,0 +1,54 @@
+
+precip_to_class <- function(precip, p.times, times.out, breaks, compress = TRUE){
+  
+  
+  if (p.times[1] != times.out[1]){
+    p.times <- c(times.out[1], p.times)
+    precip <- c(0, precip)
+  }
+  
+  if (tail(p.times, 1) != tail(times.out, 1)){
+    p.times <- c(p.times, tail(times.out, 1))
+    precip <- c(precip, tail(precip, 1))
+  }
+  precip.full <- approx(x = p.times, y = precip, xout = times.out, method = 'constant')$y
+  bins <- cut(precip.full, breaks = breaks, labels = F, right = FALSE)
+  if (compress){
+    changed <- which(as.logical(c(TRUE, diff(bins))))  
+    classes <- paste(' p-', changed, '-', bins[changed], sep = '', collapse = '')
+  } else {
+    classes <- paste(' p', seq_len(length(precip.full)), bins, sep='-', collapse = '')
+  }
+  return(paste0('precip-cell', classes))
+  
+}
+
+
+#' take the compiled-precip object and turn it into 
+#' a single sp.points.d.f w/ proper classes per timestep
+process.classified_precip <- function(viz){
+  deps <- readDepends(viz)
+  checkRequired(deps, c("compiled-precip", "timesteps", "precip-breaks"))
```

Comment:
How required is checkRequired? I didn't use it in my precip stuff yesterday? Should I be adding it back?

Reply:
it is just a good check, because if something in there is missing that you call, dep[['something']] would be null and would probably fail in a weirder way. This is a fail early assumption checker.